pkg/resource/deploy/source_eval.go

```diff
@@ -531,7 +531,12 @@ func (rm *resmon) SupportsFeature(ctx context.Context,
 	case "secrets":
 		hasSupport = true
 	case "resourceReferences":
-		hasSupport = cmdutil.IsTruthy(os.Getenv("PULUMI_EXPERIMENTAL_RESOURCE_REFERENCES"))
+		hasSupport = true
+
+		// If the experimental flag is explicitly set to a falsy value, disable this feature.
+		if v, ok := os.LookupEnv("PULUMI_EXPERIMENTAL_RESOURCE_REFERENCES"); ok && !cmdutil.IsTruthy(v) {
```

Comment:
I like the idea of keeping a flag to turn off. What do you think of renaming it to `PULUMI_DISABLE_RESOURCE_REFERENCES` to match some of our other env vars for disabling functionality (e.g. `PULUMI_DISABLE_PROVIDER_PREVIEW`, and making it truthy to disable?